lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
+    return `//${hostname}${pathname}`;
+  } else {
+    // Otherwise, it's a local path that requires a drive letter
+    var letter = pathname.codePointAt(1) | 0x20;
+    var sep = pathname[2];
+    if (letter < 96 || letter > 122 ||
+        (sep !== ':' && sep !== '|')) {
+      throw new Error('File URLs must specify absolute paths');
+    }
+    return pathname.slice(1);
+  }
+}
+
+function getPathFromURLOther(url) {
+  if (url.hostname !== '') {
+    throw new Error(
+      `File URLs on ${os.platform()} must use hostname localhost`);
+  }
+  return url.pathname;
+}
+
+function isFileUrl(path) {
+  if (typeof path !== 'string') return false;
+  if (path.length < 7) return false;
+  if ((path[0].codePointAt(0) | 0x20) !== 102) return false; // f F
+  if ((path[1].codePointAt(0) | 0x20) !== 105) return false; // i I
+  if ((path[2].codePointAt(0) | 0x20) !== 108) return false; // l L
+  if ((path[3].codePointAt(0) | 0x20) !== 101) return false; // e E
+  if (path[4] !== ':' || path[5] !== '/' || path[6] !== '/') {
+    return false;
+  }
+  return true;
+}
+
+function getPathFromURL(path) {
+  var url;
+  if (isFileUrl(path)) {
+    url = new URL(path);
+  } else if (path instanceof URL) {
+    url = path;
+  } else {
+    return path;
+  }
+  if (url.protocol !== 'file:')
```

Comment:
Maybe this can be moved into the `else if (path instanceof URL)` block, because a path that starts with `file://`(`isFileUrl `) will have the file protocol.